core/tests/karma.conf.ts

```diff
@@ -105,8 +106,14 @@ module.exports = function(config) {
     singleRun: true,
     customLaunchers: {
       Chrome_Travis: {
-        base: 'Chrome',
-        flags: ['--no-sandbox']
+        // Karma can only connect to ChromeHeadless when inside Docker.
+        base: isDocker ? 'ChromeHeadless' : 'Chrome',
+        // Discussion of the necessity of extra flags can be found here:
+        // https://github.com/karma-runner/karma-chrome-launcher/issues/154
+        // https://github.com/karma-runner/karma-chrome-launcher/issues/180
+        flags: isDocker ? ['--no-sandbox',
```

Comment:
Style nit for this codebase: If something within ( ...) or [ ... ] spans multiple lines, add a line break after the opening paren or brace.

Reply:
Done!